drivers/ethernet/eth_sam_gmac.c

```diff
@@ -227,24 +225,23 @@ static void mac_addr_set(Gmac *gmac, u8_t index,
 static int rx_descriptors_init(Gmac *gmac, struct gmac_queue *queue)
 {
 	struct gmac_desc_list *rx_desc_list = &queue->rx_desc_list;
-	struct ring_buf *rx_frag_list = &queue->rx_frag_list;
+	struct net_buf **rx_frag_list = queue->rx_frag_list;
 	struct net_buf *rx_buf;
 	u8_t *rx_buf_addr;
 
-	__ASSERT_NO_MSG(rx_frag_list->buf);
+	__ASSERT_NO_MSG(rx_frag_list);
 
 	rx_desc_list->tail = 0U;
-	rx_frag_list->tail = 0U;
 
 	for (int i = 0; i < rx_desc_list->len; i++) {
 		rx_buf = net_pkt_get_reserve_rx_data(K_NO_WAIT);
 		if (rx_buf == NULL) {
-			free_rx_bufs(rx_frag_list);
+			free_rx_bufs(rx_frag_list, rx_desc_list->len);
```

Comment:
I think we can pass `i` as length. Then there will also be no need to have `rx_frag_list[i] = NULL;` in `free_rx_bufs`.